c_src/quicer_config.c

```diff
@@ -1047,3 +958,577 @@ parse_listen_on(ErlNifEnv *env, ERL_NIF_TERM elisten_on, QUIC_ADDR *Address)
     }
   return FALSE;
 }
+
+static ERL_NIF_TERM
+get_stream_opt(ErlNifEnv *env,
+               QuicerStreamCTX *s_ctx,
+               ERL_NIF_TERM optname,
+               ERL_NIF_TERM elevel)
+{
+  QUIC_STATUS status = QUIC_STATUS_SUCCESS;
+  void *Buffer = NULL;
+  uint32_t BufferLength = 0;
+  uint32_t Param = 0;
+  QUIC_PARAM_LEVEL Level;
+  ERL_NIF_TERM res = ATOM_ERROR_NOT_FOUND;
+
+  enif_mutex_lock(s_ctx->lock);
+  if (s_ctx->is_closed)
+    {
+      enif_mutex_unlock(s_ctx->lock);
+      return ERROR_TUPLE_2(ATOM_CLOSED);
+    }
+  enif_keep_resource(s_ctx);
+  enif_mutex_unlock(s_ctx->lock);
+
+  if (!IS_SAME_TERM(ATOM_FALSE, elevel))
+    {
+      res = get_level_param(env, s_ctx->Stream, optname, elevel);
+      goto Exit;
+    }
+  else if (ATOM_QUIC_PARAM_STREAM_ID == optname)
+    {
+      uint64_t stream_id = 0;
+      Param = QUIC_PARAM_STREAM_ID;
+      Level = QUIC_PARAM_LEVEL_STREAM;
+      BufferLength = sizeof(uint64_t);
+      Buffer = &stream_id;
+    }
+  else if (ATOM_QUIC_STREAM_OPTS_ACTIVE == optname)
+    {
+      if (ACCEPTOR_RECV_MODE_PASSIVE == s_ctx->owner->active)
+        {
+          res = SUCCESS(ATOM_FALSE);
+        }
+      else
+        {
+          res = SUCCESS(ATOM_TRUE);
+        }
+      goto Exit;
+    }
+  else if (ATOM_QUIC_PARAM_STREAM_0RTT_LENGTH == optname)
+    {
+      // @TODO
+      res = ERROR_TUPLE_2(ATOM_STATUS(QUIC_STATUS_NOT_SUPPORTED));
+      goto Exit;
+    }
+  else if (ATOM_QUIC_PARAM_STREAM_IDEAL_SEND_BUFFER_SIZE == optname)
+    {
+      // @TODO
+      res = ERROR_TUPLE_2(ATOM_STATUS(QUIC_STATUS_NOT_SUPPORTED));
+      goto Exit;
+    }
+  else if (s_ctx->c_ctx)
+    {
+      res = get_connection_opt(env, s_ctx->c_ctx, optname, elevel);
+      goto Exit;
+    }
+  else
+    {
+      res = ERROR_TUPLE_2(ATOM_PARM_ERROR);
```

Comment:
Should this be `ATOM_PARAM_ERROR`? It seems consistent throughout the code, but I wonder if it was a misspelling to begin with.

Reply:
It returns `{error, param_error}`

Reply:
It is not from msquic or OTP error code or Linux errno, must be me who intro it,  I think I get the impression from `EPERM`
Good catch I will fix it.

Reply:
now I know,  it is from
```__unused_parm__``` 

but anyway `param` is better.